### PR TITLE
introduce a "gamevariant" variable to tell different IWADs apart

### PR DESCRIPTION
**Pull request by fabiangreffrath**
Chocolate Doom supports different variants that are available of the
IWADs, e.g. for Doom 2 it supports the Vanilla IWAD, the one shipped
with the Doom 3: BFG Edition as well as Freedoom: Phase 2 and FreeDM.

Each of these IWAD variants requires some specific special-casing and
the newly introduced global "gamevariant" variable can be used to keep
track of all of this.

### CODE REVIEW

File: src/doom/d_main.c [hunk 1]
```diff
@@ -812,16 +809,17 @@ void D_IdentifyVersion(void)
 
 void D_SetGameDescription(void)
 {
-    boolean is_freedoom = W_CheckNumForName("FREEDOOM") >= 0,
-            is_freedm = W_CheckNumForName("FREEDM") >= 0;
+    gamevariant = W_CheckNumForName("FREEDOOM") >= 0 ?
+                  W_CheckNumForName("FREEDM") >= 0 ?
+                  freedoom & freedm : freedoom : vanilla;
```
Comment (jmtd) [inline src/doom/d_main.c]: This is one part that I found pretty hard to parse in my head.

**Comment by jmtd** [comment]
(golfing time from me!) In general I think this is much clearer, but one thing I would ask/suggest is, why not use boolean comparisons rather than bitwise AND when checking the gamevariant variable? I guess the one reason would be if something could be more than one at a time, but the only case that might apply to is freedoom and freedm, which seems to occur only once, so could just be `gamemode == freedoom || gamemode == freedm` and still be legible. In fact that would make line d_main line 1636 (`if (gamevariant & freedoom && !(gamevariant & freedm))`) simpler too. (I also left a specific line comment)

But as I say, golfing...

**Comment by jmtd** [comment]
I think that looks great. I haven' t tested it  but it *looks* good

**Comment by fabiangreffrath** [comment]
Am Samstag, den 23.04.2016, 03:38 -0700 schrieb Jonathan Dowland:
> I think that looks great. I haven' t tested it but it looks good

Hehe, thank you!

 - Fabian

**Comment by fabiangreffrath** [comment]
Am Samstag, den 23.04.2016, 03:38 -0700 schrieb Jonathan Dowland:
> I think that looks great. I haven' t tested it but it looks good

So, what does @fragglet think?

 - Fabian

**Comment by fragglet** [comment]
Looks good. Thanks!
